### DISCUSSION TITLE
How to run ILSpy on Linux?

### DISCUSSION THREAD

**Discussion by vulns-are-features-too** [icsharpcode/ILSpy#2435] [Help]
I'm on Arch Linux and have gone through [the installation instructions for Unix](https://github.com/icsharpcode/ILSpy#unix--mac) but haven't been given a binary like `ILSpy` to run. I do see the `nupkg` file `ICSharpCode.Decompiler.Console/bin/Debug/ilspycmd.7.1.0.6488-preview1.nupkg` and I'm guessing I'm supposed to install/run that somehow?

How exactly do I run ILSpy after installation?

**Comment by adrianoc** [comment]
I don't think you can run a direct build of ILSpy on Linux.

I've been using https://github.com/icsharpcode/AvaloniaILSpy for some time and it works relatively well.

**Comment by szymor** [comment]
How is the state of affairs nowadays? I cannot see any package on AUR related to ILSpy other than AvaloniaILSpy.

**Reply by christophwille** [reply]
AvaloniaILSpy is archived (for lack of maintainers), please see #3557

**Reply by szymor** [reply]
Sure, but the question is - can ILSpy be built for Linux instead? I believe this was a major reason for maintaining AvaloniaILSpy, which is obviously discontinued.

**Reply by christophwille** [reply]
That is in the thread as well - main ILSpy would need to move to Avalonia.

**Reply by lextm** [reply]
You might monitor #3641
